File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/CourseDetailDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * DTO for comprehensive course details including reviews summary, curriculum, and related courses.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CourseDetailDTO {
    // Basic course information (from existing CouponCourseData)
    private Integer courseId;
    private String title;
    private String heading;
    private String description;
    private String author;
    private String category;
    private String subCategory;
    private String level;
    private String language;
    private Float rating;
    private Integer reviews;
    private Integer students;
    private Integer contentLength;
    private String previewImage;
    private String previewVideo;
    private String couponUrl;
    private String couponCode;
    private Integer usesRemaining;
    private java.time.Instant expiredDate;
    
    // Additional details from Udemy API
    private ReviewsSummaryDTO reviewsSummary;
    private CurriculumDTO curriculum;
    private List<RelatedCourseDTO> relatedCourses;
    private PricingInfoDTO pricingInfo;
    private IncentivesDTO incentives;
    private List<PreviewVideoDTO> previewVideos; // List of preview videos from preview page
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/CourseDetailDTO.kt
package com.thanh0x.coursedeal.dto

import com.thanh0x.coursedeal.model.coupon.CourseLevel
import java.time.Instant

/**
 * DTO for comprehensive course details including reviews summary, curriculum, and related courses.
 */
data class CourseDetailDTO(
    // Basic course information (from existing CouponCourseData)
    val courseId: Int? = null,
    val title: String? = null,
    val heading: String? = null,
    val description: String? = null,
    val author: String? = null,
    val category: String? = null,
    val subCategory: String? = null,
    val level: CourseLevel? = null,
    val language: String? = null,
    val rating: Float? = null,
    val reviews: Int? = null,
    val students: Int? = null,
    val contentLength: Int? = null,
    val previewImage: String? = null,
    val previewVideo: String? = null,
    val couponUrl: String? = null,
    val couponCode: String? = null,
    val usesRemaining: Int? = null,
    val expiredDate: Instant? = null,
    // Additional details from Udemy API
    val reviewsSummary: ReviewsSummaryDTO? = null,
    val curriculum: CurriculumDTO? = null,
    val relatedCourses: List<RelatedCourseDTO>? = null,
    val pricingInfo: PricingInfoDTO? = null,
    val incentives: IncentivesDTO? = null,
    // List of preview videos from preview page
    val previewVideos: List<PreviewVideoDTO>? = null,
)
